src/array_api_extra/_lib/_utils/_helpers.py

```diff
@@ -175,3 +176,26 @@ def asarrays(
         xa, xb = xp.asarray(a), xp.asarray(b)
 
     return (xb, xa) if swap else (xa, xb)
+
+
+def ndindex(*x: int) -> Generator[tuple[int, ...]]:
```

Comment:
It would be good to add a test to compare against numpy

Reply:
Done.